src/app/shell/personal-cabinet/provider/provider-applciations/provider-applciations.component.ts

```diff
@@ -90,8 +99,15 @@ export class ProviderApplciationsComponent extends CabinetDataComponent implemen
    * @param Application event
    */
   onReject(application: Application): void {
-    const applicationUpdate = new ApplicationUpdate(application.id, Statuses.Rejected, application?.rejectionMessage);
-    this.store.dispatch(new UpdateApplication(applicationUpdate));
+    const dialogRef = this.matDialog.open(ReasonModalWindowComponent, {
+      data: { type: ModalConfirmationType.rejectApplication },
+    });
+    dialogRef.afterClosed().subscribe((result: string) => {
```

Comment:
unsubscribe pls

Reply:
@khrMykhailo  You don't need to unsubscribe as the observable itself completes.You can verify the same by adding a finalize block to see whether observable completes itself or not.